files/edge-core/scripts/edge-core-factory-reset

```diff
@@ -17,20 +17,33 @@
 # limitations under the License.
 # ----------------------------------------------------------------------------
 
-# This is present on some, but not all gateways, and is where logs are persisted.
-rm -f /var/log/syslog*
+# Stop and remove all existing docker containers and images
+snapctl stop ${SNAP_INSTANCE_NAME}.kubelet || true
+docker stop $(docker ps -a -q) || true
+docker system prune --volumes -a -f || true
+rm -rf ${SNAP_COMMON}/var/lib/kubelet
+
+# clear logs
 journalctl --vacuum-time=1s
 
-# Stop logging until next reboot and remove previous logs
+# syslog made a copy of journalctl so we need to delete those too
 systemctl stop syslog.socket rsyslog.service
+rm -f /var/log/syslog*
+# sometimes the rm doesn't work, so let's try to truncate to 0
 find /var/log -type f -print0 | xargs -0 truncate --size=0
 
-rm -rf ${SNAP_DATA}/userdata/edge_gw_identity
-rm -rf ${SNAP_DATA}/userdata/etc
-rm -rf ${SNAP_DATA}/wigwag/etc/devicejs/devicedb.yaml
-snapctl stop ${SNAP_INSTANCE_NAME}.kubelet || true
-rm -rf ${SNAP_COMMON}/var/lib/kubelet
+# delete runtime data but preserve mbed credentials if they exist.
+# edge-core handles the mbed credentials on factory reset and it
+# has different behavior based on whether it's configured for
+# dev mode or factory mode.
+if [ -e ${SNAP_DATA}/userdata/mbed ]; then
+    cp ${SNAP_DATA}/userdata/mbed ${SNAP_COMMON}/mbed-backup
```

Comment:
Instead of this you could do something like:

```bash
find ${SNAP_DATA} -not -path "**/userdata/mbed/**" -exec rm -rf {}
```
This would prevent us from having to do the backup.